strings/Boyer_Moore_Search.py

```diff
@@ -0,0 +1,56 @@
+"""
+This algorithm tries to find the pattern in given text using Bad Character Heuristic method. 
+The bad-character rule considers the character in Text at which there is a mis-match. The next occurrence of that character to the left in Pattern is found, and a shift which brings that occurrence in line with the mismatched occurrence in Text is proposed. If the mismatched character does not occur to the left in Pattern, a shift is proposed that moves the entirety of Pattern past the point of mismatch
+
+Complexity : O(n/m)
+    n=length of main string
+    m=length of pattern string
+"""
+
+class BayerMooreSearch:
+    def __init__(self, text, pattern):
+        self.text, self.pattern = text, pattern
+        self.textLen, self.patLen = len(text), len(pattern)
+    
+
+    """ following functions tries to find the index of given char in pattern, 
+    if not found -1 will be returned"""
+    def matchInPattern(self, char):
+        for i in range(self.patLen-1, -1, -1):
+            if char == self.pattern[i]:
+                return i
+        return -1
+
+
+    """ following functions tries to find the mis-matched character's index in text from last, 
```

Comment:
Place docstrings after the "def" line.
Explain the parameters and the return value separately.

Please Google "Python docstrings conventions".
https://www.datacamp.com/community/tutorials/docstrings-python#second-head